### DISCUSSION TITLE
Ipopt doesn't give acceptable solution, although all tolerances are met

### DISCUSSION THREAD

**Discussion by Hochreiter0815** [coin-or/Ipopt#568] [Q&A] [answered]
Hi everyone,
I'm solving an OCP with IPOPT using Casadi in Matlab. During the solving, I'm not understanding why IPOPT does not give me the acceptable solution, although all limits are met. What could be the reason for this behaviour? Heres the output snippet:
```
Acceptable Check:
  overall_error = 4.0049968906621995e-004   acceptable_tol_             = 1.0000000000000000e-003
  dual_inf      = 1.8744701615457735e-003   acceptable_dual_inf_tol_    = 1.0000000000000000e+010
  constr_viol   = 1.9893289691344063e-012   acceptable_constr_viol_tol_ = 1.0000000000000000e-002
  compl_inf     = 2.7658399587907630e-007   acceptable_compl_inf_tol_   = 1.0000000000000000e+003
  curr_obj_val_ = 3.8628515676254682e-001   last_obj_val                = 3.8618856693638409e-001
  fabs(curr_obj_val_-last_obj_val_)/Max(1., fabs(curr_obj_val_)) = 9.6589826162729597e-005 acceptable_obj_change_tol_ = 1.0000000000000000e+020
test iter = 30084
Exception of type: ACCEPTABLE_POINT_REACHED in file "../../../../Ipopt/src/Algorithm/IpBacktrackingLineSearch.cpp" at line 541:
 Exception message: Restoration phase called at acceptable point.
Exception of type: LOCALLY_INFEASIBLE in file "../../../../Ipopt/src/Algorithm/IpRestoMinC_1Nrm.cpp" at line 274:
 Exception message: Restoration phase converged to a point of local infeasibility
```
Thanks.

**Comment by svigerske** [comment, marked as answer]
This seems to be inside the restoration phase (`IpRestoMinC_1Nrm.cpp` is mentioned here), so the point was only acceptable to the NLP that is solved in the restoration phase, not the one that you attempt to solve. The problem solved in the restoration phase is minimizing the constraint violation of the original NLP. Thus, the objective value 0.386285 displayed here is the norm of the constraint violation in the original NLP. Since this is larger than some threshold, the problem is concluded by the (locally) infeasible:

https://github.com/coin-or/Ipopt/blob/a62679dd2cd292a9e3aa64ad05df37f5adfdece5/src/Algorithm/IpRestoMinC_1Nrm.cpp#L278-L292

PS: The line numbers look like you use an old Ipopt version. Please consider updating.

**Reply by Hochreiter0815** [reply]
Thanks for the quick reply!
